calico_node/filesystem/etc/service/felix/run

```diff
@@ -1,5 +1,10 @@
 #!/bin/sh
 exec 2>&1
+# Felix doesn't understand HOSTNAME, but the container exports it as a common
+# interface. This ensures Felix gets the right hostname.
+if [ ! -z $HOSTNAME ]; then
+    export FELIX_FELIXHOSTNAME=$HOSTNAME
```

Comment:
That is true - and I'm happy to merge as is - that said Felix has to be resilient to empty strings for these parameters and it's unlikely that we'd change that behavior (i.e. treat "" differently from missing).

I've put up PRs to test this change in libcalico and calico-containers.